\documentclass[10pt]{beamer}
\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{helvet}
\usepackage{ragged2e}
%-------------------------------------------------------
% INFORMATION IN THE TITLE PAGE
%-------------------------------------------------------

\newcommand{\cstitle}{\textbf{Bioinformatics}}
\subtitle[]{Phylogenetic trees}
\newcommand{\cscourseCode}{1005155}
\newcommand{\csauthor}{MSc. Vicente Machaca Arceda}
\institute[UNSA]{Universidad Nacional de San Agustín de Arequipa}
\newcommand{\csemail}{vmachacaa@unsa.edu.pe}
\newcommand{\instituteabr}{UNSA}
\newcommand{\nameUp}{ICC Fase 1}
\date{\today}
\title[\cscourseCode]{\cstitle}
\author{\csauthor}
%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
% CHOOSE THE THEME
%-------------------------------------------------------
\def\mycmd{0} % CS THEME
%\def\mycmd{1} % MYTHEME
%-------------------------------------------------------


\if\mycmd1
\usetheme[]{Feather}
\newcommand{\chref}[2]{	\href{#1}{{\usebeamercolor[bg]{Feather}#2}} }
\else
\usepackage{csformat}
\fi

\newcommand{\1}{
	\setbeamertemplate{background}{
		\includegraphics[width=\paperwidth,height=\paperheight]{img/1}
		\tikz[overlay] \fill[fill opacity=0.75,fill=white] (0,0) rectangle (-\paperwidth,\paperheight);
	}
}



%-------------------------------------------------------
% THE BODY OF THE PRESENTATION
%-------------------------------------------------------

\begin{document}
	
	
	\AtBeginSubsection[]
	{
		\begin{frame}
			\frametitle{Table of Contents}
			\tableofcontents[currentsubsection]
		\end{frame}
	}
	
	
	%-------------------------------------------------------
	% THE TITLEPAGE
	%-------------------------------------------------------
	
	\if\mycmd1 % MY THEME
	\1{
		\begin{frame}[plain,noframenumbering] 
			\titlepage 
		\end{frame}}
		\else % CS THEME
		\maketitle
		\fi

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Table of Contents}
\tableofcontents
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Objectives}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Introduction}{Objectives}
\begin{itemize}
    \item<1-> Understand the importance of phylogenetic trees. 
    \item<2-> Understand and implement UPGMA.
  \end{itemize}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Phylogenetics}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Definition}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Phylogenetics}{}
	\begin{block}{What is evolution?}
		In the biological context, evolution can be defined as the development of a
		biological form from other preexisting forms or its origin to the current existing form
		through natural selections and modifications \cite{xiong2006essential}. 		
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Phylogenetics}{Definition}
	\begin{block}{}
		\textbf{Phylogenetics} is the study of the evolutionary history of living organisms using tree like diagrams to represent pedigrees of these organisms \cite{xiong2006essential}. 		
	\end{block}

	\begin{block}{}
		\textbf{Molecular phylogenetics} is the study of evolutionary relationships of genes and other biological macromolecules by analyzing mutations at various positions in their sequences \cite{xiong2006essential}. 		
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Phylogenetics}{Example}
	\begin{figure}
		\includegraphics[width=0.7\textwidth]{img/philo/example_1}
		\caption{A typical bifurcating phylogenetic tree showing root, internal nodes, terminal nodes and branches. Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Major Assumptions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Major Assumptions}{}
	Major Assumptions:	
	\begin{block}{}
		\begin{itemize}
			\item 		Molecular sequences used in phylogenetic construction are homologous, meaning that they share a common origin. 		
			\item Each position in a sequence evolved independently. 		
		\end{itemize}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Terminology}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Terminology}{}
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{img/philo/example_4}
		\caption{A phylogenetic tree showing an example of	bifurcation and multifurcation. Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Rooted and unrooted}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Rooted and unrooted}{}
	\begin{figure}
		\includegraphics[width=0.8\textwidth]{img/philo/example_5}
		\caption{An illustration of rooted versus unrooted trees. Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Rooted and unrooted}{}
	\begin{block}{}
		The root of the tree is not known; the common ancestor is already
		extinct \cite{xiong2006essential}. 		
	\end{block}
	
	\begin{block}{}
		There are two ways to define the root of a tree:
		\begin{itemize}
			\item \textbf{Outgroup}.- Which is a sequence that is homologous to the sequences under consideration, but separated from those sequences at an early evolutionary time.
			\item \textbf{Midpoint rooting approach}.- The midpoint of the two most divergent groups
			judged by overall branch lengths is assigned as the root.
		\end{itemize} 
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Gene versus species phylogeny}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Gene versus species phylogeny}{}
	\begin{block}{Gene phylogeny}
		Describes the evolution of that particular gene/protein. This sequence
		may evolve more or less rapidly than other genes or may have a different evolutionary history from the rest of the genome \cite{xiong2006essential}. 		
	\end{block}

	\begin{block}{Species phylogeny}
		The species evolution is the combined result of evolution by multiple genes in a genome \cite{xiong2006essential}. 		
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Forms of representation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Forms of representation}{cladograms and phylograms}
	\begin{figure}
		\includegraphics[width=0.7\textwidth]{img/philo/example_6}
		\caption{Phylogenetic trees drawn as cladograms (top) and phylograms (bottom). Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Forms of representation}{Newick}
	\begin{figure}
		\includegraphics[width=\textwidth]{img/philo/example_2}
		\caption{Newick format of tree representation. Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The true tree}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{The true tree}{}
	\begin{block}{}
		The	search for a correct tree topology can sometimes be extremely difficult and computationally demanding. The number of potential tree topologies can be enormously large even with a moderate number of taxa \cite{xiong2006essential}.		
	\end{block}

	\begin{equation}
		N_R = \frac{( 2n - 3)! }{ 2^{n-2} (n-2)!}
	\end{equation}
	
	\begin{equation}
	N_U = \frac{( 2n - 5)! }{ 2^{n-3} (n-3)!}
	\end{equation}
	where $N_R$ and $N_U$ are the number of rooted and unrooted trees, $n$ is the number of taxa.

\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{The true tree}{}
	\begin{figure}
		\includegraphics[width=\textwidth]{img/philo/true_tree_1}
		\caption{Unrooted and rooted trees for 3 taxa. Source: \cite{xiong2006essential}}			
	\end{figure}	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{The true tree}{}
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{img/philo/true_tree_2}
		\caption{Unrooted and rooted trees for 4 taxa. Source: \cite{xiong2006essential}}			
	\end{figure}	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{The true tree}{}
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{img/philo/true_tree_3}
		\caption{Total number of rooted ($\circ$) and unrooted ($\blacklozenge$) tree topologies as a function of the number	of taxa. The values in the y-axis are plotted in the log scale.
			Source: \cite{xiong2006essential}}			
	\end{figure}	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Methodology}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Steps}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Methodology}{Input-Output}
	\begin{columns}
		\begin{column}{0.48\textwidth}
			Distances between sequences [A, B, C, D]:
			\begin{equation*}			
				\begin{bmatrix}
				0 & 8 & 4 & 6 \\
				8 & 0 & 8 & 8 \\
				4 & 8 & 0 & 6 \\
				6 & 8 & 6 & 0 \\
				\end{bmatrix}			
			\end{equation*}
		\end{column}
		\begin{column}{0.48\textwidth}
			Output:
				\begin{figure}
					\includegraphics[width=\textwidth]{img/philo/tree}
					%\caption{Output.}			
			\end{figure}
		\end{column}
	\end{columns}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Methodology}{}
	\begin{block}{}
		\begin{itemize}
			\item Choice a molecular marker.
			\item Alignment.
			\item Multiple substitution (distances).
			\item Phylogenetics building.
		\end{itemize}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Choice of Molecular Markers}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Choice of Molecular Markers}{}
	\begin{block}{}
		Nucleotide or protein sequence data?
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Choice of Molecular Markers}{}
	Use nucleotides for:
	\begin{block}{}
		\begin{itemize}
			\item Studying very closely related organisms, nucleotide sequences, which
			evolve more rapidly than proteins.
		\end{itemize}
	\end{block}

	Use proteins because:
	\begin{block}{}
		\begin{itemize}
			\item Protein sequences are relatively more conserved as a result of the degeneracy of the	genetic code.
			\item Sixty-one codons encode for twenty amino acids, meaning thereby a change in a codon may not result in a change in amino acid.
		\end{itemize}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Choice of Molecular Markers}{}
	Moreover:
	\begin{block}{}
		\begin{itemize}
			\item Protein sequences allow more sensitive alignment than	DNA sequences (20 vs. 4 characters). \pause
			\item Two randomly related DNA sequences can result in up to 50\%
			sequence identity, compared to 10\% for protein sequences. \pause
			\item In the alignment of DNA, gaps almost always cause	frameshift errors. Protein sequences have a higher signal-to-noise ratio. \pause
			\item DNA is informative for closely related sequences. Moreover, if we take into account that  sequences evolve faster at the DNA level.
		\end{itemize}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Alignment}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Alignment}{}
	\begin{block}{}
		Only the correct alignment produces correct phylogenetic.
	\end{block}

	\begin{block}{}
		\begin{itemize}
			\item  In some cases, researchers like to remove all insertions and deletions and only use positions that are shared by all sequences in the dataset. As a consequence, many phylogenetic signals are lost.
			
			\item There is an automatic approach in improving alignment quality. For example: Rascal 	and NorMD
		\end{itemize}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Multiple Substitutions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{}
	\begin{block}{}
		After alignment, we need to measure the distance between sequences. A simple measure could be the number of substitutions in a alignment.
	\end{block}

	\begin{figure}
		\includegraphics[width=0.6\textwidth]{img/philo/align}
		\caption{This alignment has 8 substitutions, so the distance between these sequences is 8.
			Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{Homoplasy}
Substitutions are complex, for example:
	\begin{block}{}
		When $A$ is replace by $C$. The nucleotide may have actually undergone a
		number of intermediate steps to become C, such as $A 	\rightarrow T 	\rightarrow  G 	\rightarrow C$
	\end{block}

\vspace{0.5cm}
Such multiple substitutions obscure the estimation of the true evolutionary distances between sequences. This effect is known as \textbf{homoplasy} \cite{xiong2006essential}.
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{Substitution Models}

	\begin{block}{}
		The statistical models used to correct \textbf{homoplasy} are called substitution models or evolutionary models \cite{xiong2006essential}.
	\end{block}

\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Jukes–Cantor Model}	
	This model assumes that all nucleotides are substituted with equal probability.

	\begin{equation}
		d_{AB} = -\frac{3}{4} ln \left[  1 - \left( \frac{4}{3} \right) p_{AB}  \right] 
	\end{equation}\\
	
	\vspace{0.5cm}
	
	where, $d_{AB}$ is the evolutionary distance between $A$ and $B$. $p_{AB}$, is the observed sequence distance measured by the proportion of substitutions over the entire lenght. \\
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Jukes–Cantor Model}	
	\textbf{Example}: If an alignment of sequences A and B is twenty nucleotides long and
	six pairs are found to be different, the sequences differ by 30\%, or have an observed
	distance 0.3. What is the distances using Jukes–Cantor Model?		
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Jukes–Cantor Model}	
	Solution:
	
	\begin{equation*}
	d_{AB} = -\frac{3}{4} ln \left[  1 - \left( \frac{4}{3} \right) p_{AB}  \right] 
	\end{equation*}\\	
	
	$p_{AB} = 0.3$
	
	\begin{equation*}
	d_{AB} = -\frac{3}{4} ln \left[  1 - \left( \frac{4}{3} \right) 0.3  \right]  = 0.38
	\end{equation*}\\
	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Kimura Model}	
	This is a more sophisticated model in which mutation rates for transitions	and transversion are assumed to be different.
	
	\begin{equation}
	d_{AB} = -\frac{1}{2} ln (1 -  2p_{ti} - p_{tv} ) - \frac{1}{4}  ln (1 - 2p_{tv} )  
	\end{equation}\\
	
	\vspace{0.5cm}
	
	where, $d_{AB}$ is the evolutionary distance between $A$ and $B$. $p_{ti}$, is the observed
	frequency for transition and $p_{tv}$ the frequency of transversion\footnotemark. \\
	
	\footnotetext[1]{Transitions are interchanges of two-ring purines (A  G) or of one-ring pyrimidines (C  T): they therefore involve bases of similar shape. Transversions are interchanges of purine for pyrimidine bases, which therefore involve exchange of one-ring and two-ring structures.}
	
	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Kimura Model}	
	\textbf{Example}: The comparison of	sequences A and B differ by 30\%. If 20\% of changes are a result of transitions and 10\% of changes are a result of transversions. What is the distances using Kimura Model?		
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Kimura Model}	
	Solution:
	
	\begin{equation*}
	d_{AB} = -\frac{1}{2} ln (1 -  2p_{ti} - p_{tv} ) - \frac{1}{4}  ln (1 - 2p_{tv} )  
	\end{equation*}\\	

	$p_{ti} = 0.2$ \\
	$p_{tv} = 0.1$ \\
	
	\begin{equation*}
	d_{AB} = -\frac{1}{2} ln (1 -  2*0.2 - 0.1 ) - \frac{1}{4}  ln (1 - 2*0.1 )  = 0.40
	\end{equation*}\\
	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------



%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Protein sequences}	
	\begin{block}{}
		For protein sequences, the evolutionary distances from an alignment can be corrected using a PAM or JTT amino acid substitution matrix.
	\end{block}	

	For example, the Kimura model for correcting multiple substitutions in
	protein distances is:
	
	\begin{equation}
	d = -ln (1 - p - 0.2p^2)
	\end{equation}
	
	where, $p$ is the observed pairwise distance between two sequences.

\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------


%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{Among-Site Variations}
	\begin{block}{}
		In all these calculations, different positions in a sequence are assumed to be evolving
		at the same rate. However, this assumption may not hold up in reality.
	\end{block}

	\begin{block}{}
		Nevertheless:
		\begin{itemize}
			\item In DNA sequences, the rates of substitution differ for different codon positions. The	third codon mutates much faster than the other two.	
			\item For protein sequences, some amino acids change much more rarely than others.
		\end{itemize}

	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{Among-Site Variations}
	\begin{block}{}
		It has been shown that there are always a proportion of positions in a sequence
		dataset that have invariant rates and a proportion that have more variable rates.
		\textbf{The distribution of variant sites follows a $\gamma$ distribution pattern.}
	\end{block}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Multiple Substitutions}{Among-Site Variations}
	Probability curves of $\gamma$ distribution. The mathematical function of the distribution is
	$f(x) + (x^{\gamma-1} e^{-x} )/ \Gamma(\gamma)$. The curves assume different shapes depending on the $\gamma$-shape parameter ($\gamma$).
	
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{img/philo/prop}
		\caption{Probability curves of $\gamma$ distribution.
			Source: \cite{xiong2006essential}}			
	\end{figure}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}{Substitution Models}{Among-Site Variations}	
	For the Jukes-Cantor model, the evolution distance can be adjusted:
	
	\begin{equation*}
	d_{AB} = \frac{3}{4} \alpha \left[  1 - \left( \frac{4}{3}   p_{AB} \right)^{-1/\alpha} -1  \right] 
	\end{equation*}\\	
For the Kimura model, the evolutionary distance with $\gamma$ correction factor becomes:

	\begin{equation*}
	d_{AB} = \frac{\alpha}{2} \left[ (1 -  2p_{ti} - p_{tv} )^{-1/\alpha} - \frac{1}{2}  (1 - 2p_{tv} )^{-1/\alpha}  -\frac{1}{2}  \right] 
	\end{equation*}\\	
where $\alpha$ is the $\gamma$ correction factor (default = 1). 
	
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\if\mycmd1 % MY THEME
\1{
	{\1
		\begin{frame}[plain,noframenumbering]
			%\finalpage{Thank you}
			\begin{figure}[]
				\centering
				\includegraphics[width=\textwidth,height=0.7\textheight,keepaspectratio]{img/question.png}
				%\label{img:mot2}
				%\caption{Image example in 2 gray levels.}
			\end{figure}
	\end{frame}}
	\else % CS THEME
	\begin{frame}{Questions?}
		\begin{figure}[]
			\centering
			\includegraphics[width=\textwidth,height=0.7\textheight,keepaspectratio]{img/question.png}
			%\label{img:mot2}
			%\caption{Image example in 2 gray levels.}
		\end{figure}
		
	\end{frame}
	\fi
	%-------------------------------------------------------
	%-------------------------------------------------------

%-------------------------------------------------------
%-------------------------------------------------------
\begin{frame}[allowframebreaks]
	\frametitle{References}
	%\bibliographystyle{amsalpha}
	\bibliographystyle{IEEEtran}
	\bibliography{bibliography.bib}
\end{frame}
%-------------------------------------------------------
%-------------------------------------------------------

\end{document}